src/Typedefs/DependantLookup.idr

```diff
@@ -0,0 +1,25 @@
+||| This module is by itself because its only function takes ages to compile
+module DependantLookup
+
+import Data.Vect
+
+||| Lookup for maps with a shared index
+public export
+depLookup : DecEq t => {w : t} -> {f, g : t -> Type} ->
+             (vs : Vect n (v : t ** (f v, g v))) -> f w -> DecEq (f w) =>
+             Maybe (a : f w ** b : g w ** Elem (w ** (a, b)) vs)
```

Comment:
It's funny that as far as I can see, we only ever use the `b` part of the result. I guess we need the other stuff for sanity-checking.

Reply:
Ah nevermind, we use `prf` as well.